source/compute_intersections.cc

```diff
@@ -148,6 +144,10 @@ namespace dealii::NonMatching
 
     if (n_vertices_cell0 == 4 && n_vertices_cell1 == 4)
       { // rectangle-rectangle
+
+        std::swap(vertices_cell0[2], vertices_cell0[3]);
+        std::swap(vertices_cell1[2],
+                  vertices_cell1[3]); // to be consistent with dealii
```

Comment:
can you do this if it is const?

Reply:
Yes because in these lines I just swapped the vertices, and then the two vectors are given as const lvalue refs to the `compute_intersection_of_cells().`